src/main/java/com/iexec/sms/tee/session/scone/SconeSessionSecurityConfig.java

```diff
@@ -19,20 +19,22 @@
 import com.iexec.commons.poco.tee.TeeFramework;
 import com.iexec.sms.tee.ConditionalOnTeeFramework;
 import lombok.Getter;
-import org.springframework.beans.factory.annotation.Value;
-import org.springframework.context.annotation.Configuration;
+import lombok.Value;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.ConstructorBinding;
 
+import javax.validation.constraints.NotNull;
 import java.util.List;
 
-@Configuration
+@Getter
+@Value
+@ConstructorBinding
+@ConfigurationProperties(prefix = "tee.scone.attestation")
 @ConditionalOnTeeFramework(frameworks = TeeFramework.SCONE)
 public class SconeSessionSecurityConfig {
-
-    @Value("${tee.scone.attestation.tolerated-insecure-options}")
-    @Getter
-    private List<String> toleratedInsecureOptions;
-
-    @Value("${tee.scone.attestation.ignored-sgx-advisories}")
-    @Getter
-    private List<String> ignoredSgxAdvisories;
+    List<String> toleratedInsecureOptions;
+    List<String> ignoredSgxAdvisories;
+    @NotNull
+    String mode;
+    String url;
```

Comment:
It is part of the debt we will have to deal with.
URLs have a dedicated type in Java. Let's use it !
```suggestion
    URL url;
```